sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryKeyVaultSecret.Serialization.cs

```diff
@@ -29,11 +29,59 @@ void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
             writer.WriteEndObject();
         }
 
-        internal static DataFactoryKeyVaultSecret? DeserializeAzureKeyVaultSecretReference(JsonElement element)
+        void IJsonModel<DataFactoryKeyVaultSecret>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<DataFactoryKeyVaultSecret>)this).GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(DataFactoryKeyVaultSecret)} does not support writing '{format}' format.");
+            }
+
+            ((IUtf8JsonSerializable)this).Write(writer);
+        }
+
+        DataFactoryKeyVaultSecret? IJsonModel<DataFactoryKeyVaultSecret>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<DataFactoryKeyVaultSecret>)this).GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(DataFactoryKeyVaultSecret)} does not support reading '{format}' format.");
+            }
+
+            using var document = JsonDocument.ParseValue(ref reader);
+            return DeserializeAzureKeyVaultSecretReference(document.RootElement);
+        }
+
+        BinaryData IPersistableModel<DataFactoryKeyVaultSecret>.Write(ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<DataFactoryKeyVaultSecret>)this).GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(DataFactoryKeyVaultSecret)} does not support writing '{format}' format.");
+            }
+
+            return ModelReaderWriter.Write(this, options, DataFactoryContext.Default);
+        }
+
+        DataFactoryKeyVaultSecret? IPersistableModel<DataFactoryKeyVaultSecret>.Create(BinaryData data, ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<DataFactoryKeyVaultSecret>)this).GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(DataFactoryKeyVaultSecret)} does not support reading '{format}' format.");
+            }
+
+            using var document = JsonDocument.Parse(data);
+            return DeserializeAzureKeyVaultSecretReference(document.RootElement);
+        }
+
+        string IPersistableModel<DataFactoryKeyVaultSecret>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
+
+        internal static DataFactoryKeyVaultSecret DeserializeAzureKeyVaultSecretReference(JsonElement element)
         {
             if (element.ValueKind == JsonValueKind.Null)
             {
-                return null;
+                return null!;
```

Comment:
update the return type of this method to be nullable and revert the change in this line.